### DISCUSSION TITLE
Replace msiexec with Windows Installer COM interface

### DISCUSSION THREAD

**Discussion by Mrman1093** [PSAppDeployToolkit/PSAppDeployToolkit#1069] [Q&A] [answered]
The [Windows Installer COM interface](https://learn.microsoft.com/en-us/windows/win32/msi/automation-interface-reference) appears to be quite robust. Having just taken a deep dive into it, I think it would be interesting to implement more aspects of it into PSADT. I don't see any immediate benefits that would come from using this interface over msiexec in Execute-Msi, but it would make PSADT slightly more "PowerShell native" depending on whether you consider COM to be PS native. Additionally, it could be used in Remove-MsiApplications to search the MSI database rather than the ARP registry.

**Comment by DanGough** [comment]
I have plans to add the Deployment Tools Foundation libraries from the WiX project as a proof-of-concept to handle MSI duties. These have several advantages over calling the COM interfaces directly, and you get a hell of a lot of functionality exposed merely by importing the dlls.

**Reply by mjr4077au** [reply, marked as answer]
Created https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/issues/1309 on the back of this discussion. It's marked as on-hold but at least we can track it.
